packages/react/src/DateInput/DateInput.tsx

```diff
@@ -7,11 +7,20 @@ import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ForwardedRef, InputHTMLAttributes } from 'react'
 
-export type DateInputProps = InputHTMLAttributes<HTMLInputElement>
+export type DateInputProps = {
+  /** Whether the value fails a validation rule. */
+  invalid?: boolean
+} & Omit<InputHTMLAttributes<HTMLInputElement>, 'aria-invalid' | 'type'>
 
 export const DateInput = forwardRef(
-  ({ className, ...restProps }: DateInputProps, ref: ForwardedRef<HTMLInputElement>) => (
-    <input {...restProps} ref={ref} className={clsx('ams-date-input', className)} type="date" />
+  ({ className, invalid, ...restProps }: DateInputProps, ref: ForwardedRef<HTMLInputElement>) => (
+    <input
+      {...restProps}
+      aria-invalid={invalid || undefined}
```

Comment:
So the value may only be `true` or `undefined`? `false` is not an option? Then maybe give the optional `invalid?` prop above the type `true` instead of `boolean`? You either set it to `true` or you don't set it at all?

Reply:
No, `invalid` can be `true`, `false` or `undefined`. `aria-invalid` can only be `true` or `undefined`. 

Working with a boolean prop is a bit easier than working with an HTML attribute that can only be `true` or `undefined`, which is why the `invalid` prop exists.